tools/c7n_tencentcloud/c7n_tencentcloud/resources/cam.py

```diff
@@ -0,0 +1,460 @@
+        service = "cam"
+        version = "2019-01-16"
+        enum_spec = ("ListUsers", "Response.Data[]", {})
+        taggable = True
+        resource_prefix = "uin"
+        batch_size = 10
+        datetime_fields_format = {
+            "CreateTime": ("%Y-%m-%d %H:%M:%S", pytz.timezone("Asia/Shanghai")),
+            "LastLoginTime": ("%Y-%m-%d %H:%M:%S", pytz.timezone("Asia/Shanghai")),
+            "access_keys.CreateTime": ("%Y-%m-%d %H:%M:%S", pytz.timezone("Asia/Shanghai")),
+            "access_keys.LastUsedDate": ("%Y-%m-%d", pytz.timezone("Asia/Shanghai"))
+        }
+
+    def __init__(self, ctx, data):
+        super().__init__(ctx, data)
+        self._query_client = None
+
+    @property
+    def client(self):
+        """client"""
+        if self._query_client is None:
+            self._query_client = self.get_client()
+        return self._query_client
+
+    def augment(self, resources):
+        """augment"""
+        for item in resources:
+            fm = self.resource_type.datetime_fields_format["CreateTime"]
+            item["CreateTime"] = isoformat_datetime_str(item["CreateTime"], fm[0], fm[1])
+        return resources
+
+
+@User.filter_registry.register('group')
+class GroupMembership(ValueFilter):
+    schema = type_schema('group', rinherit=ValueFilter.schema)
+    schema_alias = False
+    permissions = ()
+    group_field_name = "c7n:Groups"
+
+    def get_user_groups(self, user_set):
+        """get_user_groups"""
+        for user in user_set:
+            params = {
+                # we can only create no more than 300 groups
+                "Rp": 300,
+                "SubUin": user[self.manager.resource_type.id]
+            }
+            resp = self.manager.client.execute_query("ListGroupsForUser", params)
+            user[self.group_field_name] = resp["Response"]["GroupInfo"]
+
+    def process(self, resources, event=None):
+        """process"""
+        # get users' group
+        with self.executor_factory(max_workers=2) as w:
+            futures = []
+            for user_set in chunks([r for r in resources if self.group_field_name not in r]):
+                futures.append(w.submit(self.get_user_groups, user_set))
+            for f in as_completed(futures):
+                pass
+
+        matched = []
+        for r in resources:
+            groups = r.get(self.group_field_name, [])
+            if self.data.get("value") in ["absent", None]:
+                # for below usecase:
+                #   - type: group
+                #     key: GroupName
+                #     value: null
+                if not groups:
+                    matched.append(r)
+            else:
+                for group in groups:
+                    if self.match(group) and r not in matched:
+                        matched.append(r)
+        return matched
+
+
+@User.filter_registry.register('credential')
+class CredentialFilter(Filter):
+    """CredentialFilter"""
```

Comment:
can we add an example here for the filter

Reply:
Actually we have created almost all of the policy provided by Riot.
You may refer to the file tests/test_tc_cam.py